src/bowlingGame.js

```diff
@@ -0,0 +1,56 @@
+'use strict';
+
+class BowlingGame{
+
+    constructor() {
+        this.rolls = [];
+        this.currentRoll = 0;
+    }
+
+    roll(pins) {
+        this.rolls[this.currentRoll++] = pins;
```

Comment:
If you do this then the first item in your array will be undefined as you are incrementing the value of currentRoll. An alternative could be: 

```
// populate index 0 in your array on the first roll.
this.rolls[this.currentRoll] = pins; 
// then increment currentRoll
this.currentRoll++;
```

Reply:
@TobyRet Actually, that's not true. `a++` returns the value before incrementation and then increments. `++a` increments first then returns.
Try this:
```
let a = 9
let b = a++
console.log(a)
console.log(b)
let c = ++a
console.log(a)
console.log(c)
```

Reply:
However, I'd probably put this in two lines to avoid confusion.

Reply:
Huh, that is surprising. I stand corrected! I had wrongly assumed the behaviour would be the same as Ruby ... I should have known!

```
irb(main):001:0> a = 0
=> 0
irb(main):002:0> array = []
=> []
irb(main):003:0> array[a += 1] = 'foo'
=> "foo"
irb(main):004:0> array
=> [nil, "foo"]

```

Reading up on it further, when you use the increment/decrement operator after the operand, the value will be returned before the operand is increased/decreased (as @alicelieutier mentioned). Incrementing / decrementing the operator before the operand will make the variable increment/decrement before returning.

Fun times indeed.